apps/extension/src/wallet-services.ts

```diff
@@ -6,21 +6,49 @@ import {
 } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
 import { onboardGrpcEndpoint, onboardWallet } from './storage/onboard';
 import { ServicesMessage } from './message/services';
+import { localExtStorage } from './storage/local';
+import { AppParameters } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/app/v1/app_pb';
+import { AppService } from '@penumbra-zone/protobuf';
+import { createGrpcWebTransport } from '@connectrpc/connect-web';
+import { createPromiseClient } from '@connectrpc/connect';
 
 export const startWalletServices = () =>
   backOff(
     async () => {
       const wallet = await onboardWallet();
       const grpcEndpoint = await onboardGrpcEndpoint();
+
+      const chainId =
+        (await localExtStorage
+          .get('params')
+          .then(json => json && AppParameters.fromJson(json).chainId)) ??
+        (await createPromiseClient(AppService, createGrpcWebTransport({ baseUrl: grpcEndpoint }))
+          .appParameters({})
+          .then(({ appParameters }) => appParameters?.chainId));
+
+      if (!chainId) throw new Error('No chainId available');
+
       const services = new Services({
+        chainId,
         idbVersion: IDB_VERSION,
         grpcEndpoint,
         walletId: WalletId.fromJsonString(wallet.id),
         fullViewingKey: FullViewingKey.fromJsonString(wallet.fullViewingKey),
       });
 
-      // initialize
-      await services.getWalletServices();
+      const { blockProcessor, indexedDb } = await services.getWalletServices();
+
+      // sync the IndexedDb last block number with chrome.storage.local
+      // Later used in Zustand store
```

Comment:
Probably nothing we should do now, but feels like a code-smell that we have three forms of storage for sync height that need to be in sync.